python/link_prediction.py

```diff
@@ -0,0 +1,553 @@
+        print("Remapping failed")
+        return mgp.Record(status="Preprocessing failed", metrics=[])
+     """
+
+    """ Train g is a graph which has removed test edges. Others are positive and negative train and test graphs
+    """
+
+    train_g, train_pos_g, train_neg_g, val_pos_g, val_neg_g = preprocess(graph, link_prediction_parameters.split_ratio)
+    if train_g is None or train_pos_g is None or train_neg_g is None or val_pos_g is None or val_neg_g is None:
+        print("Preprocessing failed. ")
+        return mgp.Record(status="Preprocessing failed", metrics=[])
+ 
+    # Create a model
+    model = create_model(layer_type=link_prediction_parameters.layer_type, hidden_features_size=link_prediction_parameters.hidden_features_size,
+                         aggregator=link_prediction_parameters.aggregator, attn_num_heads=link_prediction_parameters.attn_num_heads)
+    # Create a predictor
+    predictor = create_predictor(predictor_type=link_prediction_parameters.predictor_type, predictor_hidden_size=link_prediction_parameters.hidden_features_size[-1])
+
+    # Create an optimizer
+    optimizer = create_optimizer(optimizer_type=link_prediction_parameters.optimizer, learning_rate=link_prediction_parameters.learning_rate,
+                                 model=model, predictor=predictor)
+
+    # Collect training and validation results from utils model
+    training_results, validation_results = inner_train(model, predictor, optimizer,
+                                                       link_prediction_parameters.num_epochs, link_prediction_parameters.node_features_property,
+                                                       link_prediction_parameters.console_log_freq, link_prediction_parameters.checkpoint_freq,
+                                                       link_prediction_parameters.metrics, link_prediction_parameters.tr_acc_patience, link_prediction_parameters.model_save_path,
+                                                       train_g, train_pos_g, train_neg_g, val_pos_g, val_neg_g)
+
+    return mgp.Record(status="OK", training_results=training_results, validation_results=validation_results)
+
+
+@mgp.read_proc
+def predict(ctx: mgp.ProcCtx, src_vertex: mgp.Vertex, dest_vertex: mgp.Vertex) -> mgp.Record(score=mgp.Number):
+    """Predict method. We assume here semi-inductive learning process where queried nodes are somehow connected to the original graph. It is assumed that nodes are already added to the original graph and our goal
+    is to predict whether there is an edge between two nodes or not. Even if the edge exists, method can be used. 
+
+
+    Args:
+        ctx (mgp.ProcCtx): A reference to the context execution
+
+    Returns:
+        score: Probability that two nodes are connected
+    """
+    global graph, predictor, model
```

Comment:
Use id() property of vertex by calling `src_vertex.id()`. Every vertex has that property in Memgraph, and it is not related to "id" you have set in properties